### PR TITLE
Add a PL sections area to the teacher homepage

### PR DESCRIPTION
**Pull request by dmcavoy**
This PR is part of preparing for launching PL courses and being able to create PL sections.

It creates a separate table on the TeacherHomepage for PL sections that a teacher account has created. In addition it pulls out the SetUpSection component into its own separate area above both classroom sections and professional learning sections areas. The created PL sections area will only show if you have created pl section and since there are currently no pl sections (we haven't added the ability to create pl sections yet) this is safe to merge ahead of launching PL sections and courses.

Screen Shot 2022-04-13 at 10 22 08 PM

If there are no student sections and pl sections then only the Create a Section area will show.

## Links

[ENG plan](https://docs.google.com/document/d/1V61xONU0-mleMEEdHWNxhvZtYIPRCB31mT_hbUXusCQ/edit#)
[Product Spec](https://docs.google.com/document/d/1qZpc90daxQiiqpcQUiUiGV2tQoaYEQUzOA72UZzBlVE/edit#heading=h.7j0u06e8a26f)
[Jira](https://codedotorg.atlassian.net/browse/PLAT-1681)

## Testing story

- Added and update unit tests
- Manual testing locally where I have some PL sections

### CODE REVIEW

**Comment by dmcavoy** [comment]
Do not show classroom sections area unless sections
PL sections table: change grade to participant type

File: apps/src/templates/teacherDashboard/teacherSectionsRedux.js [hunk 1]
```diff
@@ -692,13 +694,24 @@ export default function teacherSections(state = initialState, action) {
...
     });
 
+    let sectionIds = _.uniq(
+      state.sectionIds.concat(sections.map(section => section.id))
+    );
+
+    let studentSectionIds = sections
+      .filter(section => section.participantType === 'student')
+      .map(section => section.id);
+    let plSectionIds = sections
```
Comment (davidsbailey) [inline apps/src/templates/teacherDashboard/teacherSectionsRedux.js]: nit: `const`, here and above

File: apps/src/templates/teacherDashboard/teacherSectionsRedux.js [hunk 2]
```diff
@@ -861,6 +876,19 @@ export default function teacherSections(state = initialState, action) {
...
+    let newSections = _.omit(state.sections, oldSectionId);
+    newSections[section.id] = {
+      ...state.sections[section.id],
+      ...section
+    };
+
+    let newStudentSectionIds = Object.values(newSections)
+      ?.filter(section => section.participantType === 'student')
+      .map(section => section.id);
+    let newPlSectionIds = Object.values(newSections)
```
Comment (davidsbailey) [inline apps/src/templates/teacherDashboard/teacherSectionsRedux.js]: `const`, here and above, including for `newSections`

File: apps/src/templates/teacherDashboard/teacherSectionsRedux.js [hunk 3]
```diff
@@ -861,6 +876,19 @@ export default function teacherSections(state = initialState, action) {
...
+      ?.filter(section => section.participantType !== 'student')
+      .map(section => section.id);
```
Comment (davidsbailey) [inline apps/src/templates/teacherDashboard/teacherSectionsRedux.js:890]: I think you can do without `?.` here and above -- `Object.values` appears to either return an array or throw an exception.

**Review by davidsbailey — APPROVED** [review]
nice work Dani! just a couple nits.

**Pull request merged**